src/lib.rs

```diff
@@ -19,10 +20,57 @@
 #[cfg(feature = "alloc")]
 extern crate alloc;
 
-#[doc(hidden)]
-pub mod _export {
-    /// A re-export of core::*
-    pub mod _core {
-        pub use core::*;
+mod error;
+mod iter;
+mod parse;
+
+#[cfg(all(feature = "alloc", not(feature = "std")))]
+use alloc::vec::Vec;
+
+use crate::parse::FromHex as _;
+
+#[rustfmt::skip]                // Keep public re-exports separate.
+#[doc(inline)]
+pub use self::{
+    error::{
+        DecodeToArrayError, DecodeToBytesError, InvalidCharError,
+        InvalidLengthError, OddLengthStringError, 
+    },
+};
+
+/// Decodes a hex string into a vector of bytes.
+///
+/// # Errors
+///
+/// Errors if `s` is not a valid hex string.
+#[cfg(feature = "alloc")]
+pub fn decode_vec(s: &str) -> Result<Vec<u8>, DecodeToBytesError> {
```

Comment:
Despite myself mentioning not referring to target types in error types, referring to them in **functions** is correct and appropriate because that's the specific type that the function returns.

However, because "to decode x" means that "x" is source, not the target, as we discussed already, this should be `decode_to_vec`.